File: Top Interview 150/3. Longest Substring Without Repeating Characters/Solution.java
class Solution {
    public int lengthOfLongestSubstring(String str) {
        Set<Character> s = new HashSet<Character>();
        int maxLen = 0;
        int leftmost = 0;
        for(int i = 0; i < str.length(); i++) {
            while(s.contains(str.charAt(i))) {
                s.remove(str.charAt(leftmost++));
            }
            s.add(str.charAt(i));
            maxLen=Math.max(maxLen, (int)s.size());
        }
        return maxLen; 
    }
}

File: Top Interview 150/3. Longest Substring Without Repeating Characters/Solution.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string str) {
        unordered_set<char> s;
        int maxLen=0;
        int leftmost=0;
        for(int i = 0; i < str.size(); i++) {
            while(s.find(str[i])!=s.end()) {
                s.erase(str[leftmost++]);
            }
            s.insert(str[i]);
            maxLen=max(maxLen, (int)s.size());
        }
        return maxLen;
    }
};
